### PR TITLE
[1LP][RFR] Turn eventing ON for Nuage provider

### PR DESCRIPTION
**Pull request by miha-plesko**
With this commit we extend Nuage provider so that it now inputs AMQP credentials upon creation as well. Until now we were only inputing the Default credentials.

Also, with this commit we update test that was only testing
- CREATE
- RETRIEVE
- DELETE

of Nuage provider to now also validate

- UPDATE

which gives us a complete CRUD test in the end.

### CODE REVIEW

File: cfme/networks/provider/__init__.py [hunk 1]
```diff
@@ -144,6 +146,76 @@ def create(self, cancel=False, validate_credentials=True, validate_inventory=Fal
...
+                            NotImplementedError(
+                                "Such endpoint doesn't have change password/key button")
+
+                        endp_view.fill(endpoint.credentials.view_value_mapping)
+                if (validate_credentials and hasattr(endp_view, 'validate') and
+                        endp_view.validate.is_displayed):
+                    endp_view.validate.click()
+                    self._post_validate_checks(edit_view)
+
+        details_view = self.create_view(navigator.get_class(self, 'Details').VIEW)
```
Comment (jamesooden) [inline cfme/networks/provider/__init__.py]: These two views are not used on cancel.   you should probably limit there scope to only where they are used.

File: cfme/networks/provider/__init__.py [hunk 2]
```diff
@@ -144,6 +146,76 @@ def create(self, cancel=False, validate_credentials=True, validate_inventory=Fal
...
+        main_view = self.create_view(navigator.get_class(self, 'All').VIEW)
+
+        if cancel:
+            edit_view.cancel.click()
```
Comment (jamesooden) [inline cfme/networks/provider/__init__.py]: Why not just have a _post_cance() routine and include the cancel click with the checks.   It seems kind of odd the separation.

File: cfme/networks/provider/__init__.py [hunk 3]
```diff
@@ -144,6 +146,76 @@ def create(self, cancel=False, validate_credentials=True, validate_inventory=Fal
...
+            self._post_cancel_edit_checks(main_view)
+        else:
+            edit_view.save.click()
```
Comment (jamesooden) [inline cfme/networks/provider/__init__.py]: Sames as 207.

Comment (jamesooden) [reply cfme/networks/provider/__init__.py]: Though this has more logic to it due to the endpoints, but I think the argument still holds.

File: cfme/networks/provider/__init__.py [hunk 4]
```diff
@@ -232,6 +321,20 @@ def step(self):
...
+    VIEW = NetworkProviderEditView
+    prerequisite = NavigateToSibling('All')
+
+    def step(self):
+        self.prerequisite_view.entities.get_entity(name=self.obj.name, surf_pages=True).check()
+        try:
+            self.prerequisite_view.toolbar.configuration.item_select('Edit Selected Network Provider')
+        except MoveTargetOutOfBoundsException:
+            # TODO: Remove once fixed 1475303
+            self.prerequisite_view.toolbar.configuration.item_select('Edit Selected Network Provider')
```
Comment (jamesooden) [inline cfme/networks/provider/__init__.py]: Perhaps a comment on why just retrying the operation is acceptable.

File: cfme/networks/provider/nuage.py [hunk 5]
```diff
@@ -69,9 +83,14 @@ def mgmt(self):
 
     @classmethod
     def from_config(cls, prov_config, prov_key, appliance=None):
```
Comment (jamesooden) [inline cfme/networks/provider/nuage.py]: Would you add a docstring here.   Every little bit helps.

File: conf/cfme_data.yaml.template [hunk 6]
```diff
@@ -180,6 +180,12 @@ management_systems:
                 credentials: nuage
```
Comment (jamesooden) [inline conf/cfme_data.yaml.template]: **REQUIRED:** These changes need to be done in the cfme_yaml repository.

File: conf/credentials.yaml.template [hunk 7]
```diff
@@ -32,6 +32,9 @@ lenovo:
 nuage:
   username: admin
   password: smartvm
+nuage_events:
```
Comment (jamesooden) [inline conf/credentials.yaml.template]: **REQUIRED:** These changes need to be done in the cfme_yaml repository.

**Comment by jamesooden** [comment]
If you look at the PRT run it's got the following error:

    ERROR: Bad signature. Please sign your commits!

I'm guess you have not gotten your gpg key stuff setup for your git commits.   If you go to the [CFME QE New Hire page](https://mojo.redhat.com/docs/DOC-1067199), and to the section "Setting up your personal GPG key." there are instructions on how to do this.   Let me know if you need any help.

File: cfme/networks/provider/nuage.py (hunk 5 above)
Comment (miha-plesko) [reply cfme/networks/provider/nuage.py]: Good point, thanks, done.

File: cfme/tests/networks/test_network_providers.py [hunk 8]
```diff
@@ -1,8 +1,10 @@
 # -*- coding: utf-8 -*-
 import pytest
+import uuid
```
[the remark this answers is not in the document]
Comment (miha-plesko) [reply cfme/tests/networks/test_network_providers.py]: Not where I copied this [from](https://github.com/manageiq/integration_tests/blob/master/cfme/tests/cloud/test_providers.py#L184) 😉 , but done. I too prefer shorter random strings over full-blown UUID.

File: cfme/networks/provider/__init__.py (hunk 4 above)
Comment (mfalesni) [reply cfme/networks/provider/__init__.py]: We had broken toolbars at one point when JS decided to be bad on us. Try if it works without the retry code, please :)

**Comment by mfalesni** [comment]
@miha-plesko I reckon that you as an external person can't have access to our internal providers, so this may get a bit tricky reviewing.

File: cfme/networks/provider/__init__.py (hunk 4 above)
Comment (jamesooden) [reply cfme/networks/provider/__init__.py]: Sure, I was hoping you knew why (-:   If not no need for a comment.   I can't fault you for those before you not leaving the reason they were dong this.

File: cfme/networks/provider/nuage.py (hunk 5 above)
Comment (jamesooden) [reply cfme/networks/provider/nuage.py]: sweet

**Comment by jamesooden** [comment]
@miha-plesko All your changes look good to me.   @psav  is going to handle the commit signing issue.   I'm going approve what I have seen, but that still leaves the [1LP] review.

**Comment by miha-plesko** [comment]
@jamesooden thank you.
@mfalesni you're right, it works now even without the ugly try-catch. I've removed it and repushed.
@psav from what I understand, this PR is now on you.

**Pull request merged**